File: JAVASCRIPTSDAS/SORTING/06SeperateEvenOdd.js
 function swap( arr , start , end  ) { 
    let temp  = arr[start] 
    arr[start] = arr[end] 
    arr[end] = temp 

 }
function SeperateEvenOdd( arr , size ) {
    let  start  = 0  , end  = size-1 ;
    while ( start < end  ) {
      if (  arr[start] % 2 == 0 ) {
          start++ ;
      }
      else if  ( arr[end] % 2 !=  0 ) {
          end --  ; 
      }
      else {
          swap(   arr ,  start , end ) ;
          start ++ ;
          end-- ;
  
      }
    }
    
}

let arr= [9,6,6,3,3,5,8,5,4,65,10]  
SeperateEvenOdd(arr , arr.length) 
console.log(arr);

File: CPPDSA/00SORTING/06SeperateEvenOdd.cpp
/* Given an array of even and odd numbers , write a program to separete even numbers from 
  odd numbers 
*/
#include<iostream>
#include<vector>
using namespace std ; 

int Separate( vector<int>&arr , int size )  {
    int start  = 0  , end  = size-1 ;
  while ( start < end  ) {
    if (  arr[start] % 2 == 0 ) {
        start++ ;
    }
    else if  ( arr[end] % 2 !=  0 ) {
        end --  ; 
    }
    else {
        swap( arr[start] , arr[end]) ;
        start ++ ;
        end-- ;

    }
  }
}

int main () {
    vector<int>arr {9,6,6,3,3,5,8,5,4,65,10} ;
    int size  = arr.size() ; 
    Separate( arr , size ) ;
     for(int val : arr  ) {
        cout<< val  <<  " " ;
     }
     return 0  ;  
}